cli/internal/scm/scm.go

```diff
@@ -51,3 +54,27 @@ func FromInRepo(repoRoot turbopath.AbsoluteSystemPath) (SCM, error) {
 	}
 	return newFallback(dotGitDir.Dir())
 }
+
+// GetCurrentBranch returns the current branch
+func GetCurrentBranch(dir string) string {
+	cmd := exec.Command("git", []string{"branch", "--show-current"}...)
+	cmd.Dir = dir
+
+	out, err := cmd.Output()
+	if err != nil {
+		return ""
+	}
+	return strings.TrimRight(string(out), "\n")
+}
+
+// GetCurrentSha returns the current SHA
+func GetCurrentSha(dir string) string {
+	cmd := exec.Command("git", []string{"rev-parse", "HEAD"}...)
+	cmd.Dir = dir
+
+	out, err := cmd.Output()
+	if err != nil {
+		return ""
+	}
+	return strings.TrimRight(string(out), "\n")
```

Comment:
```suggestion
	return strings.TrimRight(string(out), "\r\n")
```

Reply:
Will continue this thread here https://github.com/vercel/turbo/pull/4734#discussion_r1182523783